Fix TypeScript type error in Connection.query() method

## Overview

The `Connection.query()` method in the TypeScript client has a type error that prevents clean compilation.

## Error

```
src/connection.ts(123,32): error TS2339: Property 'rows' does not exist on type 'never'.
src/connection.ts(123,41): error TS7006: Parameter 'row' implicitly has an 'any' type.
```

## Location

`packages/vibesql-client-ts/src/connection.ts:123`

```typescript
return this.currentQuery.rows.map(row =>
  this.parseRow<T>(row, this.currentQuery!.columns)
);
```

## Root Cause

The `currentQuery` field type narrowing isn't working correctly. After the `if (!this.currentQuery)` check, TypeScript should narrow the type, but it's inferring `never` instead.

## Suggested Fix

Either:
1. Add explicit type annotation for `currentQuery` 
2. Store `currentQuery` in a local variable after the null check
3. Use a type guard function

## Discovery

Found during code review of PR #3870. This is a pre-existing issue on main branch.

## Related

- PR #3870 (where discovered)
- Issue #3871 (Connection class updates)

--- a/packages/vibesql-client-ts/src/connection.ts
+++ b/packages/vibesql-client-ts/src/connection.ts
@@ -120,7 +120,7 @@ export class Connection extends EventEmitter {
         return [];
       }
 
-      const queryResult = this.currentQuery;
+      const queryResult = this.currentQuery as QueryResult;  // Type assertion after null check
       return queryResult.rows.map((row: QueryRow) =>
         this.parseRow<T>(row, queryResult.columns)
       );

